### PR TITLE
Auto upgrade base image dependencies

### PR DESCRIPTION
**Pull request by jingyanwangms**
**Description**: This is for fixing docker image security vulnerabilities. With this, we don't have to upgrade every package individually.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: dockerfiles/Dockerfile.training [hunk 1]
```diff
@@ -21,11 +21,14 @@ WORKDIR /stage
 
 # install curl and git
 RUN apt-get -y update &&\
+    apt-get install -y sudo git bash unattended-upgrades && \
```
Comment (jywu-msft) [inline dockerfiles/Dockerfile.training:24]: you already install git below in line 27?
do you need the other packages that previously weren't being installed? 
can just add unattended-upgrades to the packages below

Comment (jingyanwangms) [reply dockerfiles/Dockerfile.training:24]: Just added unattened-upgrades now

[the remark this answers is not in the document]
Comment (jingyanwangms) [reply dockerfiles/Dockerfile.training:24]: It's used to upgrade the packages from nvidia base image. Without unattended-upgrade, package names has to be explicitly listed in order to be updated like in this PR: github.com/microsoft/onnxruntime-training-examples/pull/25. This is just lower maintenance

File: dockerfiles/Dockerfile.training [hunk 2]
```diff
@@ -21,11 +21,14 @@ WORKDIR /stage
...
     apt-get -y --no-install-recommends install \
         curl \
         git \
         language-pack-en
 
+RUN unattended-upgrade
```
[the remark this answers is not in the document]
Comment (jingyanwangms) [reply dockerfiles/Dockerfile.training:30]: The spelling is different install unattended-upgrades, RUN unattended-upgrade

File: dockerfiles/Dockerfile.training (hunk 1 above)
[the remark this answers is not in the document]
Comment (suffiank) [reply dockerfiles/Dockerfile.training:24]: I see, OK. (You might be missing _push_ as sudo is still there.)

File: dockerfiles/Dockerfile.training (hunk 2 above)
Comment (suffiank) [inline dockerfiles/Dockerfile.training:30]: Can you add a comment line above explaining the purpose of the command? e.g. "# force update existing packages to resolve security vulnerabilities."

File: dockerfiles/Dockerfile.training (hunk 1 above)
[the remark this answers is not in the document]
Comment (jingyanwangms) [reply dockerfiles/Dockerfile.training]: Yeah I just pushed a new commit

File: dockerfiles/Dockerfile.training (hunk 2 above)
Comment (jingyanwangms) [reply dockerfiles/Dockerfile.training:31]: It's a good idea. Added

**Pull request merged**